controllers/storagecluster/external_resources.go

```diff
@@ -340,43 +386,47 @@ func (obj *ocsExternalResources) ensureDeleted(r *StorageClusterReconciler, inst
 	return reconcile.Result{}, nil
 }
 
-// claims should be created only once and should not be created/updated again if user deletes/update it.
-func (r *StorageClusterReconciler) createDefaultStorageClassClaims(instance *ocsv1.StorageCluster) error {
-
-	storageClassClaimFile := &ocsv1alpha1.StorageClassClaim{
+// creates claims for rbd and claims should be created only once and should not be created/updated again if user deletes/update it.
+func (r *StorageClusterReconciler) createDefaultStorageClassClaimsForRBD(instance *ocsv1.StorageCluster) error {
+	storageClassClaimBlock := &ocsv1alpha1.StorageClassClaim{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      generateNameForCephFilesystemSC(instance),
+			Name:      generateNameForCephBlockPoolSC(instance),
 			Namespace: instance.Namespace,
 			Labels: map[string]string{
 				defaultStorageClassClaimLabel: "true",
 			},
 		},
 		Spec: ocsv1alpha1.StorageClassClaimSpec{
-			Type: "sharedfilesystem",
+			Type: "blockpool",
 		},
 	}
 
-	storageClassClaimBlock := &ocsv1alpha1.StorageClassClaim{
+	err := r.createAndOwnStorageClassClaim(instance, storageClassClaimBlock)
+	if err != nil {
+		return fmt.Errorf("failed to create default blockpool storageClassClaim %s. %v", storageClassClaimBlock.Name, err)
+	}
+
+	return nil
+}
+
+// creates claims for cephFs and claims should be created only once and should not be created/updated again if user deletes/update it.
+func (r *StorageClusterReconciler) createDefaultStorageClassClaimsForCephFS(instance *ocsv1.StorageCluster) error {
+	storageClassClaimFile := &ocsv1alpha1.StorageClassClaim{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      generateNameForCephBlockPoolSC(instance),
+			Name:      generateNameForCephFilesystemSC(instance),
 			Namespace: instance.Namespace,
 			Labels: map[string]string{
 				defaultStorageClassClaimLabel: "true",
 			},
 		},
 		Spec: ocsv1alpha1.StorageClassClaimSpec{
-			Type: "blockpool",
+			Type: "sharedfilesystem",
 		},
 	}
 
 	err := r.createAndOwnStorageClassClaim(instance, storageClassClaimFile)
 	if err != nil {
```

Comment:
Let's keep variables as local as we can

```suggestion
	if err := r.createAndOwnStorageClassClaim(instance, storageClassClaimFile); err != nil {
```

Reply:
done